golang/cosmos/cmd/agd/main.go

```diff
@@ -85,13 +85,22 @@ func main() {
 		args := []string{"ag-chain-cosmos", "--home", gaia.DefaultNodeHome}
 		args = append(args, os.Args[1:]...)
 
-		binary := cast.ToString(appOpts.Get(daemoncmd.FlagSplitVm))
-		if binary == "" {
-			binary, lookErr := FindCosmicSwingsetBinary()
+		vmSpec := cast.ToString(appOpts.Get(daemoncmd.FlagSplitVm))
+		var binary string
+
+		switch vmSpec {
+		case "embedded", "default":
+			var lookErr error
+			binary, lookErr = FindCosmicSwingsetBinary()
 			if lookErr != nil {
 				return lookErr
 			}
+		default:
+			// The user provided a binary path.
+			binary = vmSpec
+		}
```

Comment:
If all cases require a binary, would it not make sense to check for said binary regardless of its embedded, default, or user-provided binary?